core/src/main/java/org/apache/calcite/plan/TableAliasTrait.java

```diff
@@ -22,6 +22,16 @@
 public class TableAliasTrait implements RelTrait {
   private final String tableAlias;
 
+  private String statementType;
+
+  public String getStatementType() {
+    return statementType;
+  }
+
+  public void setStatementType(String statementType) {
```

Comment:
once set we don't want to change the statement type so instead of adding setter can we create a constructor and initialise the statementType there itself?

Reply:
done